nservicebus/azure-service-bus/addressing-logic.md

```diff
@@ -0,0 +1,171 @@
+
+### Version 7
+
+In version 7 the configuration API allows to modify all behaviours and assumptions related to the addressing logic. This article will mainly focus on describing the layers inside the addressing logic and will show how to replace parts if further changes should occur, for a full list of out of the box options refer to [Full Configuration API](/nservicebus/azure-service-bus/configuration/configuration.md).
+
+### Addressing layers
+
+The following layers are found in the addressing logic:
+
+ * Validation: Determines how entity names and paths are validated.
+ * Sanitization: Determines how invalid entity names are cleaned up.
+ * Individualization: Determines how entity names are modified when used in different consumption modes.
+ * NamespacePartitioning: Determines how entities are partitioned across namespaces.
+ * Composition: Determines how entities are hierarchically composed inside a single namespace.
+ 
+### Validation
+
+The validation layer is represented by an implementation of `IValidationStrategy`. 
+
+Out of the box there are 2 validation strategies
+ * `EntityNameValidationV6Rules`: allows letters, numbers, periods (.), hyphens (-), and underscores (-)
+ * `EntityNameValidationRules` (default): allows letters, numbers, periods (.), hyphens (-), underscores (-) and slashes (/) for queues and topics, no slashes allowed for subscriptions.
+
+The default implementation of this strategy can be replaced by using the configuration API:
+
+snippet:swap-validation-strategy
+
+#### Implementing a custom validation strategy
+
+Implementing a custom validation strategy requires a class that implements `IValidationStrategy`. This interface contains one method, called `IsValid`, which provides access to the entity path in case the entity is a queue or topic (or name if the entity is a subscription), and returns a boolean indicating if that name is invalids. The entity type is passed in as second parameter and can be used to differentiate between queues, topics and subscription.
+
+If the implementation of a validation strategy requires configuration settings, these settings can be accessed by letting NServiceBus inject the `ReadOnlySettings` into the constructor of the validation strategy.
+
+snippet:custom-validation-strategy
+
+#### Extending the configuration API for custom validation settings
+
+In order to allow configuration of the custom validation strategy it is advised to create an extension method at the `AzureServiceBusValidationSettings` extension point, returned by `.Validation()`, in the NServiceBus configuration API. This provides access to the settings, in which the value can be registered using a well known key.
+
+snippet:custom-validation-strategy-extension
+
+### Sanitization
+
+The sanitization layer is represented by an implementation of `ISanitizationStrategy`. 
+
+Out of the box there are 3 sanitization strategies
```

Comment:
2 strategies. `AdjustmentSanitization` is converted into a sample (http://docs.particular.net/samples/azure/custom-sanitization/).